syne_tune/optimizer/schedulers/multiobjective/linear_scalarizer.py

```diff
@@ -0,0 +1,160 @@
+            metric=self.single_objective_metric,
+            mode=single_objective_mode,
+            **kwargs,
+        )
+
+    def _scalarized_metric(self, result: Dict[str, Any]) -> float:
+        if isinstance(self.base_scheduler, FIFOScheduler):
+            FIFOScheduler._check_keys_of_result(result, self.metric_names())
+
+        mo_results = np.array([result[item] for item in self.metric_names()])
+        logger.debug(
+            f"Scalarizing {self.metric_names()} into {self.single_objective_metric} using linear combination"
+        )
+        return np.sum(np.multiply(mo_results, self.scalarization_weights))
+
+    def suggest(self, trial_id: int) -> Optional[TrialSuggestion]:
+        """Returns a suggestion for a new trial, or one to be resumed
+        See the docstring of the chosen base_scheduler for details
+        """
+        return self.base_scheduler.suggest(trial_id)
+
+    def on_trial_add(self, trial: Trial):
+        """Called when a new trial is added to the trial runner.
+        See the docstring of the chosen base_scheduler for details
+        """
+        return self.base_scheduler.on_trial_add(trial)
+
+    def on_trial_error(self, trial: Trial):
+        """Called when a trial has failed.
+        See the docstring of the chosen base_scheduler for details
+        """
+        return self.base_scheduler.on_trial_error(trial)
+
+    def on_trial_result(self, trial: Trial, result: Dict[str, Any]) -> str:
+        """Called on each intermediate result reported by a trial.
+        See the docstring of the chosen base_scheduler for details
+        """
+        result[self.single_objective_metric] = self._scalarized_metric(result)
+        return self.base_scheduler.on_trial_result(trial, result)
+
+    def on_trial_complete(self, trial: Trial, result: Dict[str, Any]):
+        """Notification for the completion of trial.
+        See the docstring of the chosen base_scheduler for details
+        """
+        result[self.single_objective_metric] = self._scalarized_metric(result)
+        return self.base_scheduler.on_trial_complete(trial, result)
+
+    def on_trial_remove(self, trial: Trial):
+        """Called to remove trial.
+        See the docstring of the chosen base_scheduler for details
+        """
+        return self.base_scheduler.on_trial_remove(trial)
+
+    def metric_names(self) -> List[str]:
+        """
+        :return: List of metric names.
+        """
+        return self.metric
+
+    def metric_mode(self) -> Union[str, List[str]]:
+        """
+        :return: "min" if target metric is minimized, otherwise "max".
+        """
+        return self.mode
+
+    def metadata(self) -> Dict[str, Any]:
+        """
+        :return: Metadata of the scheduler
+        """
+        return {
+            **super(LinearScalarizedScheduler, self).metadata(),
+            "scalarized_metric": self.single_objective_metric,
+        }
+
+    def is_multiobjective_scheduler(self) -> bool:
```

Comment:
I'd prefer this to become a mixin, instead of an additional method for `TrialScheduler`

Reply:
Makes sense. I will add this in a folowup review since its largely a refactor of the base TrialScheduler, not inclusion of this feature. I dont want to conflate the two